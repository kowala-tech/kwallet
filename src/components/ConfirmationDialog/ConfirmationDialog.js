import React from "react";
import PropTypes from "prop-types";
import { compose } from "recompose";
// Material UI Imports
import Button from "@material-ui/core/Button";
// Component Related Imports
import Modal from "../Modal";

class ConfirmationDialog extends React.Component {
	render() {

		const {
			open,
			title,
			content,
			rejectButtonText,
			acceptButtonText,
			onReject,
			onAccept,
		} = this.props;

		return (
			<Modal
				open={open}
				handleClose={onReject}
				title={title}
				content={content}
				rejectButton={(
					<Button onClick={onReject}>
						{rejectButtonText}
					</Button>
				)}
				acceptButton={(
					<Button onClick={onAccept}
						autoFocus
						color="secondary"
						variant="raised">
						{acceptButtonText}
					</Button>
				)}
			/>
		);
	}
}

ConfirmationDialog.propTypes = {
	open: PropTypes.bool.isRequired,
	title: PropTypes.string.isRequired,
	content: PropTypes.string.isRequired,
	rejectButtonText: PropTypes.string.isRequired,
	acceptButtonText: PropTypes.string.isRequired,
	onAccept: PropTypes.func.isRequired,
	onReject: PropTypes.func.isRequired
};

export default compose(
)(ConfirmationDialog);
