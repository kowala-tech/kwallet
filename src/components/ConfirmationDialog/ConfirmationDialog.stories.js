import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
// Story related component imports
import ConfirmationDialog from "./ConfirmationDialog";

storiesOf("Confirmation Dialog", module)
	.add("with example content", () =>
		(<ConfirmationDialog
			open={true}
			title={"Discard all changes?"}
			content={"Do you want to discard all changes?"}
			rejectButtonText={"No, keep editing"}
			acceptButtonText={"Yes, discard changes"}
			onAccept={action("onAccept")}
			onReject={action("onReject")}
		/>)
	);
