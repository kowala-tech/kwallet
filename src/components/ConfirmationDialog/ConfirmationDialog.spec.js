import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import ConfirmationDialog from "./ConfirmationDialog";

describe("ConfirmationDialog", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<ConfirmationDialog
					open={true}
					title={"Discard all changes?"}
					content={"Do you want to discard all changes?"}
					rejectButtonText={"No, keep editing"}
					acceptButtonText={"Yes, discard changes"}
					onAccept={jest.fn()}
					onReject={jest.fn()}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
