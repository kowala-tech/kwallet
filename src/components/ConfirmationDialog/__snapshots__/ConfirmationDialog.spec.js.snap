// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationDialog it should match the snapshot 1`] = `
<WithStyles(WithTheme(WithWidth(WithMobileDialog)))
  acceptButton={
    <WithStyles(Button)
      autoFocus={true}
      color="secondary"
      onClick={[MockFunction]}
      variant="raised"
    >
      Yes, discard changes
    </WithStyles(Button)>
  }
  content="Do you want to discard all changes?"
  handleClose={[MockFunction]}
  open={true}
  rejectButton={
    <WithStyles(Button)
      onClick={[MockFunction]}
    >
      No, keep editing
    </WithStyles(Button)>
  }
  title="Discard all changes?"
/>
`;
